plugins/crypto.py

```diff
@@ -3,24 +3,107 @@
+    'FROMSYMBOL': 'coin',
+    'TOSYMBOL': 'currency',
+    'PRICE': 'last',
+    'HIGH24HOUR': 'high',
+    'LOW24HOUR': 'low',
+    'MKTCAP': 'market_cap_currency',
+    'TOTALVOLUME24H': 'volume_coin',
+    'TOTALVOLUME24HTO': 'volume_currency',
+    'SUPPLY': 'supply_currency',
+}
 
-    if price_response.get("Response") == "Error":
-        return "Error: " + price_response.get("Message", "An API Error Has Occured. https://min-api.cryptocompare.com/")
 
-    if not price_response["DISPLAY"].get(fsym):
-        return "Error: Unable to lookup pricing for {}".format(fsym)
+CURRENCY_SYMBOLS = {
+    'USD': u'$',
+    'CAD': u'$',
+    'ETH': u'\u039E',
+    'BTC': u'\u20BF',
+    'RUB': u'\u20BD',
+    'JPY': u'\u00A5',
+    'CNY': u'\u00A5',
+    'EUR': u'\u20AC',
+    'AZN': u'\u20BC',
+    'ILS': u'\u20AA',
+    'THB': u'\u0E3F',
+    'KRW': u'\u20A9',
+    'KPW': u'\u20A9',
+    'VND': u'\u20AB',
+    'SEK': u'kr',
+    'ZAR': u'R'
+}
 
-    values = price_response["DISPLAY"][fsym]["USD"]
 
-    say(
-        (
-            "USD/{FROMSYMBOL}: \x0307{PRICE}\x0f - High: \x0307{HIGHDAY}\x0f "
-            "- Low: \x0307{LOWDAY}\x0f - Volume: {VOLUMEDAY}"
-            " ({VOLUMEDAYTO}) - Total Supply: {SUPPLY} - MktCap: {MKTCAP}"
-        ).format(**values)
+def get_coin_price(coin, price_basis):
+    """
+    Get the coin price against a basis price system, eg BTC to USD.
+
+    :param coin: the cryptocurrency currency to check
+    :param price_basis: the price basis to check against
+    :return: a dictionary which includes the coin price, symbols,
+             and other stats, or none if something doesn't exist
+    :rtype: dict or none
+    """
+    coin = coin.upper()
+    price_basis = price_basis.upper()
+
+    response = http.get_json(
+        'https://min-api.cryptocompare.com/data/pricemultifull',
+        query_params={
+            'fsyms': coin,
+            'tsyms': price_basis
+        }
     )
 
+    if not response or 'RAW' not in response:
+        return None
+
+    raw_coin_price = response['RAW'][coin.upper()][price_basis.upper()]
+
+    coin_price = {API_FIELD_TRANSLATION[k]: v for k, v in raw_coin_price.items() if k in API_FIELD_TRANSLATION}
+
+    if coin_price['currency'] in CURRENCY_SYMBOLS:
+        coin_price['currency_symbol'] = CURRENCY_SYMBOLS[coin_price['currency']]
+    else:
+        coin_price['currency_symbol'] = '%s ' % coin_price['currency']
+
+    return coin_price
+
+
+@hook.command(autohelp=False)
+def crypto(inp):
+    """
+    .crypto [coin] [price-basis] -
+    get the price of a given coin against an optional price basis
+    """
+    coin, currency = (inp.split(' ', 1) + [None] * 2)[:2]
+    coin = (coin or 'BTC').upper()
+    currency = (currency or 'USD').upper()
+
+    response = get_coin_price(coin, currency)
+
+    if response is None:
+        return (
+            "Conversion of {coin:s} to {currency:s} not found"
+        ).format(coin=coin, currency=currency)
+
+    return (
```

Comment:
throughout: `:,.2f` to get comma formatting for large values

`:s` should be unnecessary

`u''` is unnecessary with the new __future__.unicode_literals

Reply:
`:,.2f` - the original didn't use comma formatting but I can add it.  However, in some cases we might need more than `.2f` - if it's `<1` for the value, for instance.  I'll see what I can do on some more interesting formatting techniques.

`:s` is unnecessary, I was just being explicit.  It can be removed and implicit.

`u''` is not only unnecessary, it's not valid with Python 3 - it's currently here from a holdover from when I was writing this before the Python 3 migration.

Reply:
Also, instead of specifying `High` and `Low` - should I steal from the youtube plugin and use ` $8,138.58↑ $7219.5↓`?